presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -185,6 +185,13 @@ public static long castToLong(@SqlType(StandardTypes.DOUBLE) double value)
     }
 
     @ScalarOperator(CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long castToFloat(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return Float.floatToRawIntBits((float) value);
```

Comment:
After http://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html:

> A narrowing primitive conversion from double to float is governed by the IEEE 754 rounding rules (§4.2.4). This conversion can lose precision, but also lose range, resulting in a float zero from a nonzero double and a float infinity from a finite double. A double NaN is converted to a float NaN and a double infinity is converted to the same-signed float infinity.

So, obviously we lose precision and sometimes e.g. non-infinity double number may become _infinity_ after narrowing to _float_ but this is expected behavior as IEEE754 defines it.